services/language_translation/v2.js

```diff
@@ -21,358 +22,364 @@ module.exports = function (RED) {
+
+  function SMTNode (config) {
     RED.nodes.createNode(this, config);
     var node = this;
+ 
+    // this does nothing, but am keeping it with a commented out signature, as 
+    // it might come in handy in the future.
+    this.on('close', function() {
+    });
 
+                                                                               
+    // The node has received an input as part of a flow, need to determine 
+    // what the request is for, and based on that if the required fields 
+    // have been provided. 
     this.on('input', function (msg) {
+
+      // This declaration put here, as codeacy wants it before it is used 
+      // in the do functions below.
+      var language_translation = watson.language_translation({
+        username: username,
+        password: password,
+        version: 'v2'
+      });
+      
+      // These are var functions that have been initialised here, so that 
+      // they are available for the instance of this node to use. 
+      // Tried to make these protoypes, but couldn't get them to be 
+      // invokedd. So instead have opted to go for vars. 
+
+      // If a translation is requested, then the model id will have been 
+      // built by the calling function based on source, target and domain.
+      var doTranslate = function(msg, model_id){
+        node.status({
+          fill: 'blue',
+          shape: 'dot',
+          text: 'requesting'
+        });
+
+        // Please be careful when reading the below. The first parameter is 
+        // a structure, and the tabbing enforced by codeacy imho obfuscates
+        // the code, rather than making it clearer. I would have liked an
+        // extra couple of spaces.
+        language_translation.translate({
+          text: msg.payload,
+          model_id: model_id
+        },
+        function (err, response) {
+          node.status({})
+          if (err) {
+            node.error(err, msg);
+          } else {
+            msg.translation = {};
+            msg.translation['response'] = response;
+            msg.payload = response.translations[0].translation;
+          }
+          node.send(msg);
+        });
+      }; 
+
+      // If training is requested then the glossary will be a file input. We are using temp
+      // to sync up the fetch of the file input stream, before invoking the train service. 
+      var doTrain = function(msg, model_id, filetype){
+        node.status({
+          fill: 'blue',
+          shape: 'dot',
+          text: 'requesting training'
+        });
+
+        temp.open({
+          suffix: '.xml'
+        }, function(err, info){
+          if (!err) {
+            fs.write(info.fd, msg.payload);
+            var params = {};
+            // only letters and numbers allowed in the submitted file name
+            params.name = msg.filename.replace(/[^0-9a-z]/gi, '');
+            params.base_model_id = model_id;
+            switch (filetype) {
+              case 'forcedglossary':
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 14 space characters but found 16.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2613386480/issues/source?bid=3225115&fileBranchId=3311840#l172)